### PR TITLE
Terraform POC

### PR DESCRIPTION
**Pull request by wgu-edwin**
initial POC to make sure we can create from terraform to our vpc

### CODE REVIEW

**Comment by JohnKallies** [comment]
@wgu-edwin This looks like a good sketch to prove it works. Please get Anthony's feedback, as these CIDR blocks are huge, and nothing will be open to the public here. Document the findings in our Wiki, in a page called "Open Source AWS Account"

File: deploy/terraform/main.tf [hunk 1]
```diff
@@ -21,75 +22,72 @@ data "aws_ssm_parameter" "ami" {
...
+}
+
+resource "aws_db_instance" "osmt_db" {
+  allocated_storage    = 10
+  db_name              = "osmt_db"
+  engine               = "mysql"
+  engine_version       = "5.7"
+  instance_class       = "db.t3.micro"
+  username             = "osmt_user"
+  password             = "password"
```
Comment (JohnKallies) [inline deploy/terraform/main.tf:66]: Should this be a secret?

Comment (wgu-edwin) [reply deploy/terraform/main.tf]: yes. I have removed the string and added it as and ENV variable

File: deploy/terraform/main.tf [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+
+  # Set elasticsearch to run as a service
+  sudo systemctl daemon-reload
+  sudo systemctl enable elasticsearch
+
+  # Start the service
+  sudo systemctl start elasticsearch
+  # Check the status of the service to make sure it's running and if no print a message to the console. Printing to the console should put the
+  # message in Cloudwatch.
+  status=`curl -XGET -H "Content: application/json" "$IPV4:9200/_cluster/health?pretty" | jq -r ."status"`
```
Comment (JohnKallies) [inline deploy/terraform/main.tf:151]: Does this env var resolve?

File: deploy/terraform/variables.tf [hunk 3]
```diff
@@ -0,0 +1,36 @@
+variable "AWS_REGION" {
+  type = string
+  description = "AWS region to use for resources"
+  default = "us-east-1"
```
Comment (JohnKallies) [inline deploy/terraform/variables.tf:4]: It's OK to have this file, but values for all of these should be provided in the environment.

File: deploy/terraform/main.tf [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    protocol  = "tcp"
+    to_port   = 9200
+    cidr_blocks = [data.aws_vpc.opensource_vpc.cidr_block]
+  }
+
+  egress {
+    from_port = 0
+    protocol  = -1
+    to_port   = 0
+    cidr_blocks = ["0.0.0.0/0"]
```
Comment (JohnKallies) [inline deploy/terraform/main.tf:103]: this seems like it should be provided as well.

**Comment by JohnKallies** [comment]
You'll need to remove the delete and insert for 'flyway_schema_history'.

**Comment by JohnKallies** [comment]
I was hoping for 'test/sql' as the folder structure, as that aligns with how our internal platform repos do it.

File: deploy/terraform/main.tf [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+  ami                     = nonsensitive(data.aws_ssm_parameter.ami.value)
+  count                   = 2
+  instance_type           = "t2.micro"
+  subnet_id               = data.aws_subnet.subnet_1.id
+  vpc_security_group_ids  = [aws_security_group.elasticsearch_sg.id]
+  root_block_device {
+    volume_size = 40
+    volume_type = "gp3"
+  }
+  user_data = <<EOF
```
Comment (JohnKallies) [inline deploy/terraform/main.tf:118]: There are some env vars in this shell script that exist as if through magic. Where do these come from?

- $elasticsearch_version
- $REGION
- $ip_list
- $IPV4

**Pull request closed without merging**